server/controllers/locationDetailsController.ts

```diff
@@ -2,62 +2,57 @@ import { Request, Response } from 'express'
 import { mapHeaderNoBannerData } from '../mappers/headerMappers'
 import { Prisoner } from '../interfaces/prisoner'
 import { formatName, userHasRoles } from '../utils/utils'
-import { formatDateTimeISO } from '../utils/dateHelpers'
 import { NameFormatStyle } from '../data/enums/nameFormatStyle'
 import config from '../config'
 import { AuditService, Page } from '../services/auditService'
 import { LocationDetailsPageData } from '../interfaces/pages/locationDetailsPageData'
-import PrisonerLocationDetailsService from '../services/prisonerLocationDetailsService'
+import LocationDetailsService from '../services/locationDetailsService'
 import { OffenderBooking } from '../interfaces/prisonApi/offenderBooking'
 import logger from '../../logger'
+import { convertGroupedLocationDetails, convertLocationDetails } from './converters/locationDetailsConverter'
 
-export default class PrisonerLocationDetailsController {
+export default class LocationDetailsController {
   constructor(
-    private readonly locationDetailsService: PrisonerLocationDetailsService,
+    private readonly locationDetailsService: LocationDetailsService,
     private readonly auditService: AuditService,
   ) {}
 
-  public async displayPrisonerLocationDetails(req: Request, res: Response, prisonerData: Prisoner) {
+  public async displayLocationDetails(req: Request, res: Response, prisonerData: Prisoner) {
     const { prisonerNumber, bookingId, firstName, middleNames, lastName, prisonId } = prisonerData
-    const { clientToken } = res.locals
     const name = formatName(firstName, middleNames, lastName, { style: NameFormatStyle.firstLast })
+    const profileUrl = `/prisoner/${prisonerNumber}`
+    const { clientToken } = res.locals
 
-    const isTransfer = prisonerData.prisonId === 'TRN'
-    const isReleased = prisonerData.prisonId === 'OUT'
+    const isTransfer = prisonId === 'TRN'
+    const isReleased = prisonId === 'OUT'
     const isInactiveBooking = isTransfer || isReleased
 
     const locationDetailsLatestFirst = await this.locationDetailsService.getLocationDetailsByLatestFirst(
       clientToken,
       bookingId,
     )
 
-    const currentLocation = locationDetailsLatestFirst[0]
+    const currentLocation = !isInactiveBooking && locationDetailsLatestFirst[0]
+    const previousLocations = isInactiveBooking ? locationDetailsLatestFirst : locationDetailsLatestFirst.slice(1)
+
     const canViewCellMoveButton = userHasRoles(['CELL_MOVE'], res.locals.user.userRoles)
     const canViewMoveToReceptionButton =
       config.featureToggles.moveToReceptionLinkEnabled &&
       canViewCellMoveButton &&
       currentLocation?.location !== 'Reception'
     const receptionIsFull =
       canViewMoveToReceptionButton && (await this.locationDetailsService.isReceptionFull(clientToken, prisonId))
-
-    if (!currentLocation.assignmentEndDateTime) {
-      currentLocation.assignmentEndDateTime = formatDateTimeISO(new Date())
```

Comment:
As far as I could see, no reason to set an (incorrect) end date to the current location if it is indeed current.